pkg/kube/inject/inject.go

```diff
@@ -1052,3 +1056,41 @@ func rewriteCniPodSpec(annotations map[string]string, spec *SidecarInjectionSpec
 		}
 	}
 }
+
+// overwriteClusterInfo updates cluster name and network from url path
+// This is needed when webconfig config runs on a different cluster than webhook
+func overwriteClusterInfo(containers []corev1.Container, path string) {
+	if strings.HasPrefix(path, "/inject/") {
```

Comment:
Not sure this is right - I would just split the path, and if len>4 use the last 4 components ( assuming you use /cluster/X/net/Y ).

There is no need to hardcode the URL as /inject/ - we define it in config as a URL or base URL.
We may have an inject URL that uses a different base.

Reply:
updated, I like your suggestion of /cluster/x/net/y :) thanks!